### PR TITLE
Fix data transfer relations

### PR DESCRIPTION
**Pull request by christiancp100**
### What does it do?

Accepts components also when transferring the links. This fixes the errors related to missing relations inside components after a data transfer (export and import or transfer)

### Why is it needed?

To allow relations inside of components to be transferred

### How to test it?

Create a content type with a component that has a relation to another content type inside it and add some data to that content type. Try to transfer or export and import to another Strapi project that has the same content type and everything should be transferred, as shown in [this Loom video](https://www.loom.com/share/27a1d2627261478088f91764cc34f771?sid=33193cbe-78d0-4e1f-9180-f1c330c7b5e5)

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/16813
Fixes https://github.com/strapi/strapi/issues/16811
Fixes https://github.com/strapi/strapi/issues/16891
Fixes https://github.com/strapi/strapi/issues/16999
Fixes https://github.com/strapi/strapi/issues/17343

### CODE REVIEW

**Comment by innerdvations** [comment]
Ping @derrickmehaffy because he was interested in testing this out

**Review by innerdvations — DISMISSED** [review]
Works great! Wait for some more QA to merge but from everything I tried it seems perfect.

**Comment by derrickmehaffy** [comment]
Nevermind @christiancp100 did it already :D 

https://github.com/strapi/strapi/actions/runs/5681997107

Experimental release: `0.0.0-experimental.b8add035057504e7feb472f8cc437c83b49e7268`

**Comment by derrickmehaffy** [comment]
tested with experimental release and works perfect!!!!

image
image

Thank you very much @christiancp100 / @innerdvations

**Review by derrickmehaffy — APPROVED** [review]
LGTM tested experimental

**Review by joshuaellis — COMMENTED** [review]
Are there any tests we can implement to ensure there's no regression? Feels like it's asking to happen otherwise :)

**Comment by mrpaulphan** [comment]
@christiancp100 I love u

**Comment by derrickmehaffy** [comment]
> 0.0.0-experimental.b8add035057504e7feb472f8cc437c83b49e7268

That is a good idea but we would need to be able to spin up two apps at the same time to run a transfer and then do the e2e tests on the destination (same honestly for export/import but just wiping the database between).

I think that's a bit more complex than what should be in this PR as we have quite a few EE customers and Cloud customers waiting on this fix so assuming everything passes QA we should get this one merged and released ASAP and add some tests after.

**Comment by innerdvations** [comment]
> Are there any tests we can implement to ensure there's no regression? Feels like it's asking to happen otherwise :)

@christiancp100 We might be able to just take an existing test for entities and make sure the test data has relations and components, which would be good enough to test this specific case.

On top of that, we've got an item in our backlog for setting up DTS e2e tests, thanks to the changes from the playwright tests for giving a better testing foundation. That should help ensure the entire process stays functional from now on.

**Comment by joshuaellis** [comment]
> We might be able to just take an existing test for entities and make sure the test data has relations and components, which would be good enough to test this specific case.

Yeah i thought something like this would be good enough ✨ thanks folks

**Review by Convly — APPROVED** [review]

**Comment by Convly** [comment]
_(waiting for the test, but LGTM :tada:)_

**Comment by Convly** [comment]
> tested with experimental release and works perfect!!!!
> 
> image image
> 
> Thank you very much @christiancp100 / @innerdvations

Very random but isn't "transfered" a typo? Shouldn't it be "transferred" instead? :sweat_smile:

**Comment by christiancp100** [comment]
> > tested with experimental release and works perfect!!!!
> > image image
> > Thank you very much @christiancp100 / @innerdvations
> 
> Very random but isn't "transfered" a typo? Shouldn't it be "transferred" instead? 😅

Yes 😂

**Comment by innerdvations** [comment]
> > 
> > Very random but isn't "transfered" a typo? Shouldn't it be "transferred" instead? 😅
> 
> Yes 😂

https://github.com/strapi/strapi/pull/17491 👀

**Comment by EsGeh** [comment]
Can we count on this being in an official release some time soon?
Impatiently waiting for this...

**Review by Convly — DISMISSED** [review]

**Review by Convly — APPROVED** [review]

File: packages/core/data-transfer/src/engine/__tests__/engine.test.ts
Comment (CodeBleu) [inline packages/core/data-transfer/src/engine/__tests__/engine.test.ts:43]: - Should line 39 be manyToOne?
- Not sure if line 34 needs to be singular, or if line 35 needs to be plural.  

I'm not a typescript person, just noticing some differences :smile:

**Comment by mohammed-shameem** [comment]
Is this fix now available in any latest versions ?

**Pull request merged**
